lego/apps/events/views.py

```diff
@@ -220,14 +221,29 @@ def create(self, request, *args, **kwargs):
         serializer.is_valid(raise_exception=True)
         username = serializer.data['username']
 
+        if not get_permission_handler(Event).has_perm(request.user, 'EDIT', obj=reg.event):
+            raise PermissionDenied()
+
+        try:
+            user = User.objects.get(username=username)
+        except User.DoesNotExist:
+            raise ValidationError({
+                'error': f'There is no user with username {username}',
+                'error_code': 'no_user'
+            })
+
         try:
-            reg = self.get_queryset().get(user__username=username)
+            reg = self.get_queryset().get(user=user)
         except Registration.DoesNotExist:
-            raise ValidationError({'error': 'No user with that username'})
+            raise ValidationError({
+                'error': 'The registration does not exist',
+                'error_code': 'not_registered'
```

Comment:
Make these codes constants? :)

Reply:
Yeeeeah, we could, but it seemed kind of premature, so I figured we could just inline them. 

In the long run we might want to define a set of errors, shared between the back and front-end such that we can handle errors correctly on the front-end without weird string parsing stuff.